dojo/filters.py

```diff
@@ -1354,12 +1358,6 @@ def __init__(self, *args, **kwargs):
             self.form.fields[
                 'test__engagement__product__prod_type'].queryset = Product_Type.objects.filter(
                 authorized_users__in=[get_current_user()])
-            self.form.fields[
-                'test'].queryset = Test.objects.filter(
-                Q(engagement__product__authorized_users__in=[get_current_user()]) |
-                Q(engagement__product__prod_type__authorized_users__in=[get_current_user()]))
-        # str() uses test_type
-        self.form.fields['test'].queryset = self.form.fields['test'].queryset.prefetch_related('test_type')
```

Comment:
I think this should stay

Reply:
IIRC, each entry in the drop-down was triggering an extra DB hit, but that might have been addressed by one of the `select_related` or `prefetch_related` additions.

I'll revisit. We had a large number of tests, and this drop-down was originally problematic... but we reduced the number of tests in our DB.

Reply:
I think removing it might not be bad idea, but I beleive the performance hit was solved. I remember because it was a pain to find out where the root cause was :-)

Reply:
The branch just pushed to this PR has this added back in.